include/seqan3/alignment/matrix/detail/alignment_score_matrix_one_column.hpp

```diff
@@ -0,0 +1,172 @@
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::alignment_score_matrix_one_column.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/matrix/detail/alignment_matrix_column_major_range.hpp>
+#include <seqan3/alignment/matrix/detail/alignment_score_matrix_one_column_base.hpp>
+#include <seqan3/alignment/matrix/detail/alignment_score_matrix_proxy_type.hpp>
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/simd/concept.hpp>
+#include <seqan3/std/iterator>
+#include <seqan3/std/ranges>
+#include <seqan3/std/span>
+
+namespace seqan3::detail
+{
+
+/*!\brief An alignment score matrix storing only a single column for the computation.
+ * \tparam score_t The type of the score; must model either seqan3::Arithmetic or seqan3::detail::Simd.
+ * \ingroup alignment_matrix
+ *
+ * \details
+ *
+ * This implementation allocates only a single column for the alignment score matrix.
+ * The matrix allows access to the underlying values through a range based interface. An iterator over the score matrix
+ * iterates in column-major-order. Dereferencing an iterator returns a view over the current matrix column.
+ * The value type is seqan3::detail::alignment_score_matrix_proxy_type, which gives an unified access to the respective
+ * matrix cells as needed by the standard alignment algorithm. The matrix is modelled as std::ranges::InputRange since
+ * the alignment algorithm iterates only once over the complete matrix to calculate the values.
+ */
+template <typename score_t>
+class alignment_score_matrix_one_column :
+    protected alignment_score_matrix_one_column_base<score_t>,
+    public alignment_matrix_column_major_range<alignment_score_matrix_one_column<score_t>>
+{
+private:
+
+    static_assert(Arithmetic<score_t> || Simd<score_t>,
+                  "The score type must be either an arithmetic type or a simd vector type.");
+
+    /*!\name Auxiliary types
+     * \{
+     */
+    //!\brief The base class for data storage.
+    using matrix_base_t = alignment_score_matrix_one_column_base<score_t>;
+    //!\brief The base class for iterating over the matrix.
+    using range_base_t = alignment_matrix_column_major_range<alignment_score_matrix_one_column<score_t>>;
+
+    //!\brief The actual element type.
+    using element_type = typename matrix_base_t::element_type;
+    //!\copydoc seqan3::detail::alignment_matrix_column_major_range::alignment_column_type
+    using alignment_column_type = typename range_base_t::alignment_column_type;
+    //!\copydoc seqan3::detail::alignment_matrix_column_major_range::column_data_view_type
+    using column_data_view_type = std::span<element_type>;
```

Comment:
Well, in this context it is not necessary, since the underlying container is not a generic user input but a fixed type that is part of the classes invariant.